### PR TITLE
Opt-out mechanism for emails

### PR DESCRIPTION
**Pull request by flozia**
# References: 
**Jira**
- [MNTOR-1239](https://mozilla-hub.atlassian.net/browse/MNTOR-1039): Endpoint to get unsubscribe link
- [MNTOR-1290](https://mozilla-hub.atlassian.net/browse/MNTOR-1290): Migrate Opt-out mechanism for monthly summary emails to V1

# Description

Migrates the existing flow for users to unsubscribe from emails. Currently, we only support unsubscribing from the monthly unresolved breach reports.

# Screenshot

pr-2889-screenshot

# How to test

- Visit `/preview/emails` and select the “Monthly unresolved breaches” template
- Click on “unsubscribe” in the email footer
- You should be directed to `/user/unsubscribe-monthly` and see a message that you successfully unsubscribe
- If the query param `token` is missing the user should be redirected to `/`

# Checklist (Definition of Done)
- [x] Localization strings (if needed) have been added.
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).
- [ ] I've added or updated the relevant sections in readme and/or code comments
- [x] I've added a unit test to test for potential regressions of this bug.
- [x] Product Owner accepted the User Story (demo of functionality completed) or waived the privilege.
- [x] All acceptance criteria are met.
- [x] Jira ticket has been updated (if needed) to match changes made during the development process.
- [x] Jira ticket has been updated (if needed) with suggestions for QA when this PR is deployed to stage.

### CODE REVIEW

File: src/utils/email.js [hunk 1]
```diff
@@ -75,58 +77,139 @@ function sendEmail (recipient, subject, html) {
...
 
-  url = appendUtmParams(url, campaign, content)
-  url.searchParams.append('token', encodeURIComponent(subscriber.primary_verification_token))
+async function postUnsubscribe (req, res) {
+  const data = req.body
+  const { token, utm_campaign: utmCampaign } = data
 
-  return url
+  try {
+    switch (utmCampaign) {
```
Comment (Vinnl) [inline src/utils/email.js:181]: Re your question in Slack, I would indeed not use this - some people use [extensions](https://connect.mozilla.org/t5/ideas/add-clearurls-to-firefox-for-android-mobile/idc-p/17262) or [browsers](https://support.brave.com/hc/en-us/articles/9982188779405-What-does-Copy-clean-link-mean-) that strip these.

File: src/client/css/partials/unsubscribe.css [hunk 2]
```diff
@@ -0,0 +1,6 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+.unsubscribe {
```
Comment (flozia) [inline src/client/css/partials/unsubscribe.css:5]: There is currently no specific design in the current version of Monitor and also no spec for the refresh. I’ll sync with UX if there is anything they’d like to add.

File: src/client/js/components/toast-alert.js [hunk 3]
```diff
@@ -109,6 +117,11 @@ const html = `
 </output>
 `
 
+const ToastTypes = {
+  Error: 'error',
+  Success: 'success'
+}
```
Comment (flozia) [inline src/client/js/components/toast-alert.js:123]: Adding additional types to the `toast-alerts`. Could potentially have `warning` or others in the future.

File: src/controllers/unsubscribe.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+import { guestLayout } from '../views/guestLayout.js'
+import {
+  unsubscribe,
+  unsubscribeMonthly
+} from '../views/partials/unsubscribe.js'
+import { generateToken } from '../utils/csrf.js'
+import { unsubscribeFromMonthlyReport } from '../utils/email.js'
+
+function unsubscribePage (req, res) {
```
Comment (flozia) [inline src/controllers/unsubscribe.js:13]: Adding the general unsubscribe page back in. We will only direct users to this view when we re-instantiate the option to unsubscribe from all emails.

File: src/client/js/partials/unsubscribe.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const unsubscribeButton = document.querySelector('.js-unsubscribe-button')
+
+unsubscribeButton.addEventListener('click', async event => {
```
Comment (flozia) [inline src/client/js/partials/unsubscribe.js:7]: This already adds the ability to unsubscribe from the general unsubscribe view so we only have to make `unsubscribeFromEmails` in `src/utils/email.js` functional again.

File: src/routes/user.js [hunk 6]
```diff
@@ -29,4 +36,14 @@ router.get('/settings', requireSessionUser, settingsPage)
...
 
+// unsubscribe from emails
+router.get('/unsubscribe', urlEncodedParser, asyncMiddleware(unsubscribePage))
+router.post('/unsubscribe', asyncMiddleware(unsubscribeFromEmails))
+
+router.get(
+  '/unsubscribe-monthly',
+  urlEncodedParser,
+  asyncMiddleware(unsubscribeMonthlyPage)
+)
```
Comment (flozia) [inline src/routes/user.js:47]: Directly unsubscribe users that follow the “unsubscribe” link from the monthly emails.

**Review by rhelmer — APPROVED** [review]

**Pull request merged**